.github/workflows/plugins.yml

```diff
@@ -0,0 +1,115 @@
+name: plugins
+
+on:
+  # Automatically run CI on Release and Pre-Release tags and main branch
+  # (only if there are changes to relevant paths)
+  push:
+    tags:
+      - "xp-engine-plugin/v[0-9]+.[0-9]+.[0-9]+*"
+    branches:
+      - main
+    paths:
+      - ".github/workflows/plugins.yml"
+      - "plugins/**"
+
+  # Automatically run CI on branches, that have active PR opened
+  pull_request:
+    branches:
+      - main
+    paths:
+      - ".github/workflows/plugins.yml"
+      - "plugins/**"
+
+jobs:
+  lint-go:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Go
+        id: setup-go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.16
+      - uses: actions/checkout@v3
+      - name: Lint Plugins module
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.41.1
+          working-directory: plugins/turing
+          skip-pkg-cache: true
+          skip-build-cache: true
+          args: --timeout 3m --verbose
+
+  turing-experiment-engine-xp-plugin-tests:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./plugins/turing
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Run Plugins test
+        run: make test-plugin
+
+  release-rules:
+    runs-on: ubuntu-latest
+    outputs:
+      release-type: ${{ steps.release-rules.outputs.release-type }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: release-rules
+        uses: ./.github/actions/release-rules
+        with:
+          prefix: xp-engine-plugin/
+
+  publish-turing-experiment-engine-xp-plugin:
+    # Automatically publish release and pre-release artifacts.
+    #
+    # As for dev releases, make it possible to publish artifacts
+    # manually by approving 'deployment' in the 'manual' environment.
+    #
+    # Dev build can be released either from the 'main' branch or
+    # by running this workflow manually with `workflow_dispatch` event.
+#    if: >-
```

Comment:
Currently commented to trigger workflow via PR. Will be uncommented once PR is approved prior to merging.